CDP4Reporting/DataSource/CategoryHierarchy.cs

```diff
@@ -56,6 +56,17 @@ public class Builder
             /// </summary>
             private CategoryHierarchy current;
 
+            /// <summary>
+            /// Initializes a new instance of the <see cref="Builder"/> class.
+            /// </summary>
+            /// <param name="iteration">
+            /// The <see cref="Iteration"/> containing the desired <see cref="CDP4Common.SiteDirectoryData.Category"/> items.
+            /// </param>
+            public Builder(Iteration iteration)
```

Comment:
Seemed more logical in when writing the Avalon (user) code. Not sure if the end version will be this way though.

Reply:
I agree that it does indeed seem more user friendly this way, and the user code from Avalon is cleaner-looking, but then we need to handle the case where the user intentionally creates an empty CategoryHierarchy. As far as I see, in the current implementation it would crash with an unhandled exception.